### ISSUE TITLE
Conflicts with outputReferences when custom transform changes token value

### ISSUE THREAD

**Issue by germain-gg** [amzn/style-dictionary#974]
I am using [tokens.studio](https://tokens.studio/) and their set of style-dictionary transforms https://github.com/tokens-studio/sd-transforms

Specifically a transform called `ts/resolveMath`. It appears that the transform is applied before the formatting phase. And one part of the system gets confused and the output is invalid CSS. See a more in-depth [investigation by the package maintainer](https://github.com/tokens-studio/sd-transforms/issues/13#issuecomment-1486981956)

## Reproduction steps

```jsonc
{
  "space": {
    "scale": {
      "value": "4",
      "type": "spacing",
      "description": "Base unit for spacing scale."
    },
    "6x": {
      "value": "{space.scale} * 6",
      "type": "spacing"
    },
    "7x": {
      "value": "{space.scale} * 7",
      "type": "spacing"
    }
  }
}
```

And a style-dictionary config of

```js
{
  "transformGroup": "css-tokens",
  "transforms": [
    "ts/resolveMath",
    "ts/size/px",
    "name/cti/kebab"
  ],
  "files": {
    "destination": "style.css",
    "format": "css/variables",
    "options": {
      "outputReferences": true
    }
  }
}
```

### Expected output

```css
:root {
  --space-scale: 4px;
  --space-7x: 28px;
  --space-6x: 24px;
}
```

### Actual output

```css
:root {
  --space-scale: 4px;
  --space-7x: 28px;
  --space-6x: 2var(--space-scale);
}
```

**Comment by chazzmoney** [comment]
Thanks for your note.

After looking into it, there is indeed a conflict between ts/resolveMath and "outputReferences": true. Using both of these will cause problems because they both work functionally on strings.

For the output you wish to see, I would suggest removing the "outputReferences": true. This should create the output that you want. If, however, you wanted something more like:

```
:root {
  --space-scale: 4px;
  --space-7x: var(--space-scale) * 7;
  --space-6x: var(--space-scale) * 6;
}
```

Then you can leave in "outputReferences": true and remove the ts/resolveMath transform.

In general these two systems both intend to process the strings to create different outputs, so should not be used together.

**Comment by chazzmoney** [comment]
@jorenbroekema if you have additional thoughts, happy to hear them

**Comment by jorenbroekema** [comment]
Another related issue here https://github.com/tokens-studio/sd-transforms/issues/121 where I share my thoughts.

But basically, the regex replacement that outputReferences does on the current value doesn't work if your value has been transformed, e.g. by means of a transitive transform like a math resolver or [something simpler like demonstrated here](https://configurator.tokens.studio/#project=jVPBbtswDP0VQZc0mWNjh10yZJcVPQ7DulsdDLJNp1ptyZCooIHhf68o1Y6dtkB9MCTy8ZGPpHpealXLY/rfasV3vM8VYzm32pkScr5jD2Qg02aTbVLUT6BsAOecPIckBnSNwFqb1lJMP8aUdn4nCxqh7AR8GB3kUqKFrESZPUEhipwnc2d7/uks6vbvGB/T03dILvSdgVo+E7XXUM04cl442VS/BT5Gb7hmvsBsgaplA9el9ZcjQSqwKJVAST0gqn8nYaQofGAaBCdLPFUrMEIp34R+g9QdkS5bNvkcdg7/QA0GVBlrRONgjhsul+l4iIdw97+BJ9zAUVoEc39751QZUvqJ+uHLttMG2T2eG7iVwSPMmdVGt2xlybqtJvPqe65ydYVNR+5pTjdBCZ47IP0n0fiSo26a9+790QZ/WBWJ8gRRaTD6TpaPYHbsJqzimu1/sHBKiY7t93tfabW1nShhWwizulAR9VXkL9cWYNJOGAt3jRYYfWkoc82+sK8+fFh7pb5tFl0l9fwJzN4L5Vusfq311eq/aifJ3xY7R815XVlPI9Vx3O0hGdm8ko/Z+pA99RmHT/JOyzC8AA==)

At first, my suggestion was also that outputReferences and transforms that alter the value of a token, cannot be combined, however, I came up with an idea that I'd like to explore:

It might work better if the replace done by style-dictionary was done on the original value, like `value = original.value.replace(ref, 'var(--foo)')` where `ref = "{bar}"` and `original.value = "{foo}"`.

Although that would mean for a math expression value like "{foo} + {bar}" you would get the outcome of var(--foo) + var(--bar), which you would need to wrap in a calc() statement for it to work in CSS, but hey that's kind of doable because we can scan the outputted value and determine if it's a math expression, and wrap it in a calc() statement if so, using a custom CSS format? Anyway, with regards to math expressions, this is something we are still actively discussing, whether it belongs inside the design token format, whether as a string, an AST-like object, or just as something you do in a step before you get your tokens JSON. So I wouldn't worry too much about the math edge case... rather focus on the "simpler demo" I linked :)
